equinix/data_source_network_device.go

```diff
@@ -638,6 +638,9 @@ func dataSourceNetworkDeviceRead(ctx context.Context, d *schema.ResourceData, m
 	uuid := uuidIf.(string)
 
 	validDeviceStatusList, err := getNeDeviceStatusList(d.Get(neDeviceSchemaNames["ValidStatusList"]).(string))
+	if err != nil {
+		return diag.Errorf("cannot get network device status list due to '%v'", err)
```

Comment:
added this because it showed up in my linter that `err` was not being checked.

Likewise for the simple typo fixes.